### DISCUSSION TITLE
[Apple ARM - NVIDIA CUDA] Apply a condition to install a feature

### DISCUSSION THREAD

**Discussion by noe-sautel** [devcontainers/community#78] [Q&A]
Hello,
I'm facing a challenge on a Mac with an M2 chip. The issue relates to installing the NVIDIA CUDA toolkit, specifically, the container image: ghcr.io/devcontainers/features/nvidia-cuda:1.

The problem I'm encountering is that NVIDIA no longer supports Mac OS development for this toolkit, as indicated in their [official documentation](https://developer.nvidia.com/nvidia-cuda-toolkit-11_7_0-developer-tools-mac-hosts). 
Note that, I have other developers on my repository who are using containers with a classic chip architecture, and it's essential for me to retain this feature for them.

My primary question is: How can I add a condition to specify the architecture and install or not the feature?

I've been exploring possible solutions, one in bash with a postCreateCommand to check the architecture I'm hitting a roadblock in achieving this. Any guidance, insights, or pointers would be greatly appreciated.

**Comment by samruddhikhandale** [comment]
Hi 👋 

Unfortunately, the Dev Container Spec does not have this option for optionally ignoring a Feature based on architecture/condition.
Feel free to add this proposal to the spec repo.

However, as https://developer.nvidia.com/nvidia-cuda-toolkit-11_7_0-developer-tools-mac-hosts is official, we can update the [nvidia-cida](https://github.com/devcontainers/features/tree/main/src/nvidia-cuda) Feature to skip installation of the toolkit for Apple M2s. 

@noe-sautel If you are interested, feel free to contribute a PR with these changes, thanks!

**Reply by noe-sautel** [reply]
Thank you for your quick reply and informations @samruddhikhandale 🙏 I opened a PR:[https://github.com/devcontainers/features/pull/730](https://github.com/devcontainers/features/pull/730)
